src/main/java/org/veupathdb/service/eda/ds/plugin/AbstractPlugin.java

```diff
@@ -102,7 +102,7 @@
     validateVisualizationSpec(_pluginSpec);
 
     // get list of data streams required by this subclass
-    _requiredStreams = getRequestedStreams(_pluginSpec);
+    _requiredStreams = _requiredStreams == null ? _requiredStreams = getRequestedStreams(_pluginSpec) : _requiredStreams;
```

Comment:
Not sure what you're doing here (and with the protected above).  There should still only be one call to get the required streams, yes?  Just overriding to force compute viz classes to implement the two-arg version?

Reply:
yes. thats what im trying to do anyway. i kind of figured youd pick that out.. and was prepared to say something like 'ryan its just until we have a compute service'

Reply:
if youve got a better short term solution than that, feel free, obviously

Reply:
Don't understand why the original line here was not sufficient- looks like you are caching the value in case you need to ask for streams multiple times.  But we aren't do that, are we?

Reply:
I think you're probably right. How to handle the requested streams was something I changed my mind about a couple times. But the way I have it now I think we don't need this change.